src/client/pubsub.rs

```diff
@@ -263,39 +257,32 @@ impl PubsubConnectionInner {
             }
         }
 
-        Ok(true)
+        Ok(())
     }
 
     /// Returns true, if there are still valid subscriptions at the end, or false if not, i.e. the whole thing can be dropped.
-    fn handle_messages(&mut self, cx: &mut Context) -> Result<bool, error::Error> {
+    fn handle_messages(&mut self, cx: &mut Context) -> Result<(), error::Error> {
         loop {
             match self.connection.poll_next_unpin(cx) {
-                Poll::Pending => return Ok(true),
+                Poll::Pending => return Ok(()),
                 Poll::Ready(None) => {
-                    if self.subscriptions.is_empty() {
-                        return Ok(false);
-                    } else {
-                        // This can only happen if the connection is closed server-side
-                        for sub in self.subscriptions.values() {
-                            sub.unbounded_send(Err(error::Error::Connection(
-                                ConnectionReason::NotConnected,
-                            )))
-                            .unwrap();
-                        }
-                        for psub in self.psubscriptions.values() {
-                            psub.unbounded_send(Err(error::Error::Connection(
-                                ConnectionReason::NotConnected,
-                            )))
-                            .unwrap();
-                        }
-                        return Err(error::Error::Connection(ConnectionReason::NotConnected));
+                    // This can only happen if the connection is closed server-side
```

Comment:
Here I removed the special case handling, because I couldn't think of a reason to keep it. Am I missing anything?